include/arch/armv7a/arch.h

```diff
@@ -31,8 +31,8 @@
 #define __MEMMOVE
 
 
-#if __ARM_PCS_VFP || (__VFP_FP__ && !__SOFTFP__)
-#if __ARM_FP & 8
+#if defined(__ARM_PCS_VFP) || (defined(__VFP_FP__) && !defined(__SOFTFP__))
```

Comment:
Not really the same as was, what if e.g. `__ARM_PCS_VFP` is set to zero? Not sure if this might be the case, from what I see the use case seen on github is to do it the same way you did it.

Reply:
Looking in practice
```
$ echo | arm-phoenix-gcc -dM -E  -mcpu=cortex-m4 -mfloat-abi=soft - | grep __ARM_PCS_VFP
$ echo | arm-phoenix-gcc -dM -E -mcpu=cortex-m7 -mfloat-abi=hard -mfpu=fpv5-d16 - | grep __ARM_PCS_VFP
#define __ARM_PCS_VFP 1
```

According to https://developer.arm.com/documentation/dui0774/g/chr1383660321827, I believe that this variable can be set only to 1, when specific condition is met (target is 32-bit and PCS is compatible with floating point)

I believe that it's true for (almost?) all compiler feature defines – it's not defined when not available, defined to 1 when available